src/sql/workbench/contrib/query/browser/fileQueryResultsWriter.ts

```diff
@@ -0,0 +1,336 @@
+import { ILogService } from 'vs/platform/log/common/log';
+import { Progress } from 'vs/platform/progress/common/progress';
+import { IEditorService } from 'vs/workbench/services/editor/common/editorService';
+import { IUntitledTextEditorService } from 'vs/workbench/services/untitled/common/untitledTextEditorService';
+
+export class FileQueryResultsWriter implements IQueryResultsWriter {
+	private messages: Array<IQueryMessage> = [];
+	private formattedQueryResults: Array<string> = [];
+	private tables: Array<Table<any>> = [];
+	private numQueriesToFormat: number = 0;
+	private queryContainsError: boolean = false;
+	private closingMessageIncluded: boolean = false;
+	private hasCreatedResultsFile: boolean = false;
+	private runner: QueryRunner;
+
+	constructor(
+		runner: QueryRunner,
+		@IInstantiationService private instantiationService: IInstantiationService,
+		@IUntitledTextEditorService private readonly untitledEditorService: IUntitledTextEditorService,
+		@IEditorService private readonly editorService: IEditorService,
+		@ILogService private readonly logService: ILogService,
+		@IConfigurationService private configurationService: IConfigurationService
+	) {
+		this.runner = runner;
+	}
+
+	public onQueryStart() {
+		this.reset();
+	}
+
+	public onResultSet(resultSet: ResultSetSummary | ResultSetSummary[]) {
+		this.numQueriesToFormat++;
+		let resultsToAdd: ResultSetSummary[];
+		if (!Array.isArray(resultSet)) {
+			resultsToAdd = [resultSet];
+		} else {
+			resultsToAdd = resultSet.splice(0);
+		}
+
+		if (this.configurationService.getValue<IQueryEditorConfiguration>('queryEditor').results.streaming) {
+			this.addResultSet(resultsToAdd);
+		} else {
+			resultsToAdd = resultsToAdd.filter(e => e.complete);
+			if (resultsToAdd.length > 0) {
+				this.addResultSet(resultsToAdd);
+			}
+		}
+	}
+
+	public async updateResultSet(resultSet: ResultSetSummary | ResultSetSummary[]) {
+		let resultSetSummaries: ResultSetSummary[];
+		if (!Array.isArray(resultSet)) {
+			resultSetSummaries = [resultSet];
+		} else {
+			resultSetSummaries = resultSet.splice(0);
+		}
+
+		if (this.configurationService.getValue<IQueryEditorConfiguration>('queryEditor').results.streaming) {
+			for (let resultSetSummary of resultSetSummaries) {
+				let table = this.tables.find(t => t.resultSet.batchId === resultSetSummary.batchId && t.resultSet.id === resultSetSummary.id);
+				if (table) {
+					await table.updateResult(resultSetSummary);
+				} else {
+					this.logService.warn('Got result set update request for non-existant table');
+				}
+			}
+		} else {
+			resultSetSummaries = resultSetSummaries.filter(e => e.complete);
+			if (resultSetSummaries.length > 0) {
+				this.addResultSet(resultSetSummaries);
+			}
+		}
+	}
+
+	public async onMessage(incomingMessage: IQueryMessage | IQueryMessage[]) {
+		if (isArray(incomingMessage)) {
+			incomingMessage.forEach(m => {
+				if (m.message.includes('Started executing query')) {
```

Comment:
I guess we don't localize this string?  if we do, we should localize here as well, or find a different way to suppress this message.